src/Subscription.js

```diff
@@ -56,13 +60,25 @@ export default class Subscription extends EventEmitter {
         /** * Message handlers ** */
 
         this.on('unsubscribed', () => {
+            Object.keys(this.gaps).forEach((gap) => {
+                clearInterval(gap)
+            })
             this.setResending(false)
         })
 
         this.on('disconnected', () => {
             this.setState(Subscription.State.unsubscribed)
+            Object.keys(this.gaps).forEach((gap) => {
+                clearInterval(gap)
+            })
             this.setResending(false)
         })
+
+        this.on('error', () => {
+            Object.keys(this.gaps).forEach((gap) => {
+                clearInterval(gap)
+            })
```

Comment:
This piece of code is repeated a lot and should probably be factored out e.g. `stopAllGapPolling()` or something. You should also `delete this.gaps[gap]` otherwise you got a mem leak.